Objects/longobject.c

```diff
@@ -220,11 +221,16 @@ _PyLong_FromMedium(sdigit x)
 {
     assert(!IS_SMALL_INT(x));
     assert(is_medium_int(x));
-    /* We could use a freelist here */
-    PyLongObject *v = PyObject_Malloc(sizeof(PyLongObject));
+
+    // We use _Py_FREELIST_POP_MEM instead of _Py_FREELIST_POP because the new
+    // reference is created in _PyObject_Init
+    PyLongObject *v = (PyLongObject *)_Py_FREELIST_POP_MEM(ints);
```

Comment:
Can you go back to `_Py_FREELIST_POP` and avoid reinitializing the object on line 237. We don't need to assign the `ob_type` field in objects taken from the freelist. Something like:

```C
    PyLongObject *v = (PyLongObject *)_Py_FREELIST_POP(ints);
    if (v == NULL) {
        v = PyObject_Malloc(sizeof(PyLongObject));
        if (v == NULL) {
            PyErr_NoMemory();
            return NULL;
        }
        _PyObject_Init((PyObject*)v, &PyLong_Type);
    }
    digit abs_x = x < 0 ? -x : x;
    _PyLong_SetSignAndDigitCount(v, x<0?-1:1, 1);
    v->long_value.ob_digit[0] = abs_x;
    return (PyObject*)v;
```